### PR TITLE
htop: 2.02 -> 2.1.0

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/system/htop/default.nix [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
 
   src = fetchurl {
-    sha256 = "11zlwadm6dpkrlfvf3z3xll26yyffa7qrxd1w72y1kl0rgffk6qp";
     url = "http://hisham.hm/htop/releases/${version}/${name}.tar.gz";
+    sha256 = "0j07z0xm2gj1vzvbgh4323k4db9mr7drd7gw95mmpqi61ncvwq1j";
   };
 
   buildInputs =
-    [ ncurses ] ++
+    [ ncurses python ] ++
```
Comment (dotlambda) [inline pkgs/tools/system/htop/default.nix]: Isn't it sufficient to put this in `nativeBuildInputs`?

File: pkgs/tools/system/htop/default.nix [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
     lib.optionals stdenv.isDarwin [ IOKit ];
 
+  patchPhase = ''
```
Comment (dotlambda) [inline pkgs/tools/system/htop/default.nix]: I think it's better to use `post/prePatch`.

File: pkgs/tools/system/htop/default.nix [hunk 3]
```diff
@@ -1,19 +1,23 @@
 { lib, fetchurl, stdenv, ncurses,
-IOKit }:
+IOKit, python }:
 
 stdenv.mkDerivation rec {
   name = "htop-${version}";
-  version = "2.0.2";
```
Comment (andir) [inline pkgs/tools/system/htop/default.nix]: Your commit message states `2.02`. Can you please change that to 2.0.2?

File: pkgs/tools/system/htop/default.nix (hunk 2 above)
Comment (andir) [reply pkgs/tools/system/htop/default.nix]: Yes, either is fine. I always tend to use `prePatch`. You should usually not touch the `patchPhase` unless really required. If we ever need the standard `patches` attribute with this derivation not having to rename other attributes would be great :-)

**Review by andir — CHANGES_REQUESTED** [review]

File: pkgs/tools/system/htop/default.nix (hunk 1 above)
Comment (adisbladis) [reply pkgs/tools/system/htop/default.nix]: Indeed 👍
It's only used for build time code generation

**Comment by magnetophon** [comment]
@dotlambda Thanks for the feedback.  Done.

**Comment by adisbladis** [comment]
Thanks! :)

**Pull request merged**